lib/base/view/base_view.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flutter/material.dart';
+import 'package:githubdummy/base/view_model/base_view_model.dart';
+import 'package:githubdummy/widgets/app_bar_view.dart';
+import 'package:githubdummy/widgets/loading_view.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+
+import '../../config/themes/theme.dart';
+
+abstract class BaseView<T extends BaseViewModel> extends StatelessWidget {
+  final T viewModel;
+  final Widget? child;
+
+  const BaseView({
+    super.key,
+    required this.viewModel,
+    this.child,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider<T>(
+      create: (context) => viewModel,
+      child: Consumer<T>(
+        builder: _buildScreen,
+      ),
+    );
+  }
+
+  Widget _buildScreen(BuildContext context, T viewModel, Widget? child) {
+    if (viewModel.errorMessage != null) {
+      return _buildError(context, viewModel.errorMessage!);
+    }
+
+    if (!viewModel.isInitialized) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    return _buildSafeArea(context, viewModel);
+  }
+
+  Widget _buildSafeArea(BuildContext context, T viewModel) {
+    return SafeArea(
+      left: safeAreaLeft(),
+      top: safeAreaTop(),
+      right: safeAreaRight(),
+      bottom: safeAreaBottom(),
+      minimum: safeAreaMinimumPadding(),
+      maintainBottomViewPadding: maintainBottomViewPadding(),
+      child: wrapScaffoldWithSafeArea()
+          ? _buildScaffold(context, viewModel)
+          : _buildScaffoldBody(context, viewModel),
+    );
+  }
+
+  Widget _buildScaffold(BuildContext context, T viewModel) {
+    return Scaffold(
+      appBar: _buildAppBar(context),
+      body: _buildScaffoldBody(context, viewModel),
+      bottomNavigationBar: buildBottomNavigationBar(context),
+    );
+  }
+
+  Widget _buildScaffoldBody(BuildContext context, T viewModel) {
+    return viewModel.isLoading
+        ? const Center(
+            child: LoadingView(),
+          )
+        : buildScreen(context);
```

Comment:
Try to overlay loading view onto screen. Don't replace the view. It's not a good user experience.

Reply:
Implemented with Stack and Visibility

``` dart
return Stack(
      children: [
        buildScreen(context),
        Visibility(
          visible: viewModel.isLoading,
          child: const LoadingView(),
        ),
      ],
    );
```